src/tribler/core/components/ipv8/eva_protocol.py

```diff
@@ -453,43 +497,52 @@ async def on_data(self, peer, payload):
     def send_acknowledgement(self, peer, transfer):
         transfer.acknowledgement_number = transfer.block_number + 1
 
-        logger.debug(f'Acknowledgement ({transfer.acknowledgement_number}). Window size: {transfer.window_size}. '
-                     f'Peer hash: {hash(peer)}')
+        logger.debug(f'Ack ({transfer.acknowledgement_number}). Window size: {transfer.window_size}. Peer: {peer}')
 
         acknowledgement = Acknowledgement(transfer.acknowledgement_number, transfer.window_size, transfer.nonce)
         self.community.eva_send_message(peer, acknowledgement)
 
     async def on_error(self, peer, payload):
         message = payload.message.decode('utf-8')
-        logger.debug(f'On error. Peer hash: {hash(peer)}. Message: "{message}"')
+        logger.debug(f'On error. Peer: {peer}. Message: "{message}"')
         transfer = self.outgoing.get(peer, None)
```

Comment:
**nitpick:** `None` is not necessary

Reply:
**question:** Are we safe from the situation where the error was for the previous transfer of the same peer, but got delayed somehow? Maybe the error message should include the transfer's `nonce` to be safe?

Reply:
Yes, it is indeed safer to add `nonce`, but then we will break backward compatibility.

Probably for Tribler, it is not dangerous, because we don't use this handler:

https://github.com/Tribler/tribler/blob/1f11f16481b36a1844dec42af0152ddd478b3bc2/src/tribler/core/components/metadata_store/remote_query_community/remote_query_community.py#L165-L166

So, I see only one drawback: if somewhere in the future we will want to join super-app and Tribler, then, in this case, we will have some inconsistencies in the protocol, which will need to be resolved.